lib/browser-auto-tracking.js

```diff
@@ -87,6 +87,122 @@ export default function initAutoTrackingCore(lib) {
       cookie.set('initialReferrer', initialReferrer, cookieDomain);
     }
 
+    //todo (clarity): should all this code live somewhere else?
+    //todo (clarity): setup sampling
+    if (options.clarity_sample_rate > 0.0) {
+      const microsoft_project_id = "XXXXXXXXXX";  //not needed, example "f579s7kgqn"
+      //todo (clarity): how will this endpoint look in production?
+      const clarity_upload_endpoint = `https://clarity-initial-test.datapipe.prodperfect.com/clarity-initial-test-0/${session_uuid}`;
+      //todo (clarity): how are we going to manage/update this (store in our own S3?)
+      const clarity_tracking_code = "https://www.clarity.ms/s/0.7.8/clarity.js";
+
+      //todo (clarity): how do I make directly calling this code work?
+      const clarity_loading_script = `
+        !(function (c, l, a, r, i, t, y) {
+          if (a[c].v || a[c].t) return a[c]("event", c, "dup." + i.projectId);
+          (a[c].t = !0),
+              ((t = l.createElement(r)).async = !0),
+              (t.src = clarity_tracking_code),
+              (y = l.getElementsByTagName(r)[0]).parentNode.insertBefore(t, y), a[c]("start", i),
+              a[c].q.unshift(a[c].q.pop());
+        })("clarity", document, window, "script", {
+          projectId: microsoft_project_id,
+          upload: clarity_upload_endpoint,
+          expire: 365,
+          cookies: ["_uetmsclkid", "_uetvid"],
+          track: true,
+          lean: false,
+          content: true,
+          extract: [3, 1, ["4f509md79", "zjy56jaz", "4s4zm72hj"]],
+        });
+      `;
+
+      !(function(c,l,a,r,i,t,y){
+        c[a]=c[a]||function(){
+          (c[a].q=c[a].q||[]).push(arguments)};
+          t=l.createElement(r);
+          t.async=1;
+          t.text=clarity_loading_script;
+          y=l.getElementsByTagName(r)[0];
+          y.parentNode.insertBefore(t,y);
+      })(window, document, "clarity", "script", microsoft_project_id);
+
+      //todo (clarity): how should we actually get/set cookies? (this is taken from clarity)
+      function getCookie(key) {
+        let cookies = document.cookie.split(';');
+        if (cookies) {
+          for (let i = 0; i < cookies.length; i++) {
+            let pair = cookies[i].split('=');
+            if (pair.length > 1 && pair[0] && pair[0].trim() === key) {
+              return pair[1];
+            }
+          }
```

Comment:
Maybe you can't use this for compatibility reasons but figured I'd mention this exists
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/get

Reply:
So, my hope is to replace this whole `getCookie` business with a library, preferably one that's already installed (like directly calling it from the Clarity install, somehow?).  I'm not really worried about browser compatibility stuff.  However, I do need to make sure it will likely get the same value as Clarity, so that when I set it later it works right.  Given that Clarity has all this weirdo logic about picking the right domain, I'm unclear how to align that? I assume higher-level pre-existing libraries might just handle this similarly?  I poked around with the installed `js-cookie` (comes with Keen), but that's both opinionated and definitely doesn't handle the domain stuff the same (I had 3 copies of the same cookie, none of which agreed).   Also, I can't actually get `browser.cookies.get` to run in my console, so I can't even try it out properly (I know, that's a me issue).